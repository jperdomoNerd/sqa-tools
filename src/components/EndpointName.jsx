const EndpointName = ({ endpointUrl }) => {
  return (
    <div className="endpoint-name">
      <label className="endpoint-name__label" htmlFor="endpoint-name">Endpoint:</label>
      <input className="endpoint-name__input" id="endpoint-name" type="text" value={endpointUrl} readOnly />
    </div>
  );
}

export default EndpointName;
