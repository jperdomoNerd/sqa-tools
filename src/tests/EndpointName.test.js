import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import EndpointName from '../components/EndpointName';

describe('EndpointName component', () => {
  
  it('<EndpointName /> renders correctly', () => {
    const tree = renderer.create(<EndpointName />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Label exits', () => {
    render(<EndpointName />);
    const label = screen.getByLabelText('Endpoint:');
    expect(label).toBeInTheDocument();
  });

});
