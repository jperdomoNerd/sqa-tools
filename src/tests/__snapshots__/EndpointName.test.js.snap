// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndpointName component <EndpointName /> renders correctly 1`] = `
<div
  className="endpoint-name"
>
  <label
    className="endpoint-name__label"
    htmlFor="endpoint-name"
  >
    Endpoint:
  </label>
  <input
    className="endpoint-name__input"
    id="endpoint-name"
    readOnly={true}
    type="text"
  />
</div>
`;
